### PR TITLE
[tests] add debug logging

### PR DESCRIPTION
**Pull request by wgtdkp**
Add more verbose test output to help debug when test cases fail.

### CODE REVIEW

**Review by bukepo — APPROVED** [review]

File: tests/integration/common.sh [hunk 1]
```diff
@@ -220,15 +220,33 @@ form_network() {
     set -e
     local pskc=$1
 
+    echo "------ forming Thread network ------"
+
+    echo "sudo ${OT_CTL} channel ${CHANNEL}"
```
Comment (simonlingoogle) [inline tests/integration/common.sh:225]: Being unfamiliar with the reason of adding these logging, should we use `set -x` instead ?

Comment (wgtdkp) [reply tests/integration/common.sh]: changed to use `set -x`.

File: tests/integration/run_tests.sh [hunk 2]
```diff
@@ -62,7 +64,7 @@ run_test_case() {
     ## we cannot declare output with `local`,
     ## because `local` is a command and its return value iwll override
     ## return value of the test case.
-    output=$(${test_case})
+    $(set -xeuo pipefail && ${test_case} &> "${output_file}")
```
Comment (wgtdkp) [inline tests/integration/run_tests.sh:67]: This is where we enables "xeuo pipefail" for all test cases.

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Review by jwhui — APPROVED** [review]

**Pull request merged**